tests/drivers/adc/adc_api/boards/nucleo_u083rc.overlay

```diff
@@ -0,0 +1,25 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 STMicroelectronics
+ */
+
+/ {
+	zephyr,user {
+		/* adjust channel number according to pinmux in board.dts */
+		io-channels = <&adc1 0>;
```

Comment:
You defined two ADC pins in your dts. Use both in the overlays, that way we test sampling multiple channels in one go.
To repeat on your other overlay.